src/cli/linker.zig

```diff
@@ -265,6 +279,38 @@ pub fn link(allocs: *Allocators, config: LinkConfig) LinkError!void {
             try args.append("/ignore:4217"); // Ignore locally defined symbol imported warnings
             try args.append("/ignore:4049"); // Ignore locally defined symbol imported warnings
         },
+        .freestanding => {
+            // WebAssembly linker (wasm-ld) for freestanding wasm32 target
+            try args.append("wasm-ld");
+
+            // Add output argument
+            try args.append("-o");
+            try args.append(config.output_path);
+
+            // Don't look for _start or _main entry point - we export specific functions
+            try args.append("--no-entry");
+
+            // Export all symbols (the Roc app exports its entrypoints)
+            try args.append("--export-all");
+
+            // Disable garbage collection to preserve host-defined exports (init, handleEvent, update)
+            // Without this, wasm-ld removes symbols that aren't referenced by the Roc app
+            try args.append("--no-gc-sections");
```

Comment:
@rtfeldman -- I've made an assumption here that this is the right approach for a WASM static library. We may need to revisit the design for other kinds of static libraries too for how they expose symbols.